pkg/user/tx_client.go

```diff
@@ -390,6 +412,11 @@ func (client *TxClient) broadcastTx(ctx context.Context, txBytes []byte, signer
 	if err != nil {
 		return nil, err
 	}
+	if resp.TxResponse == nil {
+		// This case should ideally not happen with BroadcastMode_SYNC if err is nil,
+		// but handle defensively.
+		return nil, fmt.Errorf("broadcast via %s returned nil TxResponse despite nil error", conn.Target())
+	}
```

Comment:
[question] what motivated this addition? Is it actually possible to have a nil TxResponse and a nil err?

Reply:
Thanks for explaining. Seems safe to keep.